Add docs for monorepo (#2045)
vercel/docs: pages/docs/v2/git-integrations/index.mdx

<Image
  src={`${process.env.ASSETS}/docs/common/production-branch-configuration.png`}
  width={1488 / 2}
  height={536 / 2}
  shadow
  borderRadius
/>
<Caption>Configuring the Production Branch of your Project.</Caption>

## Monorepos

You can have multiple Projects that are connected to the same Git repository.
Every branch push and merge will trigger deployments for all the connected projects.

To start, you can create all your Projects by [importing the Git repository](/docs/v2/git-integrations#quick-installation) multiple times.

> Each time you can select a Root Directory of one Project. It's also allowed to use the same Root Directory for multiple Projects.

Reviewer: ```suggestion
Each time you can select the Root Directory of one Project. It's also allowed to use the same Root Directory for multiple Projects.
```

<Image
  src={`${process.env.ASSETS}/docs/common/monorepo-import.png`}
  width={1284 / 2}
  height={966 / 2}
  shadow
  borderRadius
/>
<Caption>Selecting the Root Directory of your Project.</Caption>

> Once all the Projects are created, they're all connected to the same monorepo.

Reviewer: ```suggestion
Once all Projects are created, they'll all be connected to the same monorepo.
```

> Every push will now automatically create Deployments for all the Projects, as well as comment on Pull Requests and commits with status updates.

Reviewer: ```suggestion
Every push will now automatically create Deployments for all Projects, as well as comment on Pull Requests and commits with status updates.
```
